.travis.yml

```diff
@@ -0,0 +1,30 @@
+language: rust
+cache: cargo
+rust:
+    - 1.35.0
+    - stable
+    - beta
+    - nightly
+os:
+    - linux
+    - osx
+matrix:
+    allow_failures:
+        - rust: nightly
+
+before_script:
+    - |
+      (travis_wait rustup component add rustfmt-preview || true) &&
+      (test "$TRAVIS_RUST_VERSION" == 1.35.0 || travis_wait rustup component add clippy-preview || true)
+script:
+    - |
+      export PATH="$PATH":~/.cargo/bin &&
+      export RUST_BACKTRACE=1 &&
+      export CARGO_INCREMENTAL=1 &&
+      cargo build &&
+      cargo test --release &&
+      cargo test --release -- --ignored &&
+      cargo doc --no-deps &&
+      (test "$TRAVIS_RUST_VERSION" != nightly || cargo test --all --release --benches) &&
+      (test "$TRAVIS_RUST_VERSION" == 1.35.0 || cargo clippy -- --deny clippy::all) &&
+      (test "$TRAVIS_RUST_VERSION" == 1.35.0 || cargo fmt --all -- --check)
```

Comment:
Again, all 3 probably don't need the conditions (except for the benchmarks, but that one can probably be just left out completely, because we don't have any benchmarks yet).

Also, i'd propose calling clippy as:

```
cargo clippy --all --tests -- --deny clippy::all
```

That'll also check lints in test code and any potential sub-crates if we do workspaces in the future.

Reply:
I understand, thank you, addressed in fcc2d6634d80879785c7027e478d77f70f8db915